server/src/main/java/org/elasticsearch/search/aggregations/support/TimeSeriesIndexSearcher.java

```diff
@@ -90,7 +96,7 @@ public void search(Query query, BucketCollector bucketCollector) throws IOExcept
             }
             Scorer scorer = weight.scorer(leaf);
             if (scorer != null) {
-                LeafWalker leafWalker = new LeafWalker(leaf, scorer, bucketCollector, () -> tsidOrd[0]);
+                LeafWalker leafWalker = new LeafWalker(leaf, scorer, minimumScore, bucketCollector, () -> tsidOrd[0]);
```

Comment:
What about wrapping the scorer here with a `MinScoreScorer` wrapper?
The `DocIdSetIterator` it produces in the `LeafWalker` constructor only returns documents that have at least the minimum score.

This way `LeafWalker` class doesn't need to be changed and there shouldn't be any overhead for users not using the minimum score feature?